modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/filter/TimeRangeFilterHooks.es.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import React, {useCallback, useMemo, useState} from 'react';
+
+import Filter from '../../shared/components/filter/Filter.es';
+import {useFilterFetch} from '../../shared/components/filter/hooks/useFilterFetch.es';
+import {useFilterName} from '../../shared/components/filter/hooks/useFilterName.es';
+import {useRouterParams} from '../../shared/hooks/useRouterParams.es';
+import {
+	parseQueryDate,
+	formatDescriptionDate
+} from '../process-metrics/util/timeRangeUtil.es';
+import {CustomTimeRangeForm} from './CustomTimeRangeForm.es';
+
+const isCustomFilter = filter => filter.key === 'custom';
+
+const onChangeFilter = selectedFilter => {
+	const preventDefault = isCustomFilter(selectedFilter);
+
+	return preventDefault;
+};
+
+const TimeRangeFilter = ({
+	className,
+	dateEndKey = 'dateEnd',
+	dateStartKey = 'dateStart',
+	dispatch,
+	filterKey = 'timeRange',
+	options: {
+		hideControl = false,
+		multiple = false,
+		position = 'left',
+		withSelectionTitle = false
+	} = {}
```

Comment:
Define options property with a default value and remove destruction and passing options "{...options}" to Filter component.